File: app/src/main/java/com/cyl/musiclake/ui/music/download/DownloadContract.java
package com.cyl.musiclake.ui.music.download;


import com.cyl.musiclake.base.BaseContract;
import com.cyl.musiclake.bean.Music;
import com.cyl.musiclake.data.download.TasksManagerModel;

import java.util.List;

public interface DownloadContract {

    interface View extends BaseContract.BaseView {
        void showErrorInfo(String msg);

        void showSongs(List<Music> musicList);

        void showDownloadList(List<TasksManagerModel> modelList);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {
        void loadDownloadMusic();

        void loadDownloading();
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/download/ui/DownloadContract.kt
package com.cyl.musiclake.ui.download.ui


import com.cyl.musiclake.ui.base.BaseContract
import com.music.lake.musiclib.bean.BaseMusicInfo
import com.cyl.musiclake.ui.download.TasksManagerModel

interface DownloadContract {

    interface View : BaseContract.BaseView {
        fun showErrorInfo(msg: String)

        fun showSongs(baseMusicInfoInfoList: List<BaseMusicInfo>)

        fun showDownloadList(modelList: List<TasksManagerModel>)
    }

    interface Presenter : BaseContract.BasePresenter<View> {
        fun loadDownloadMusic(isCache: Boolean)

        fun loadDownloading()
    }
}
